src/format.ts

```diff
@@ -0,0 +1,82 @@
+import type { MessageEntity } from './deps.deno.ts'
+
+export interface Stringable {
+  toString(): string;
+};
+
+class FormattedString {
+  text: string
+  entities: MessageEntity[]
+
+  constructor(text: string, entities: MessageEntity[]) {
+    this.text = text
+    this.entities = entities
+  }
+
+  toString() {
+    return this.text;
+  }
+}
+
+const unwrap = (stringLike: Stringable): FormattedString => {
```

Comment:
Is there a reason why you're avoiding functions so radically?

Reply:
Not really. Consistency? Is there a reason to use function declaration - perhaps in situation where we need function declaration hoisting?

Reply:
I just find it more readable. A function is a function, a const is a const. If a function needs to be assigned to a const, there's usually a good reason, hence the question. But it's a nit, this isn't blocking.